azurerm/internal/services/postgres/postgresql_flexible_server_configuration_test.go

```diff
@@ -0,0 +1,146 @@
+	data := acceptance.BuildTestData(t, "azurerm_postgresql_flexible_server_configuration", "test")
+	r := PostgresqlFlexibleServerConfigurationResource{}
+	name := "backslash_quote"
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.template(data, name, "on"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("name").HasValue(name),
+				check.That(data.ResourceName).Key("value").HasValue("on"),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.empty(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				data.CheckWithClientForResource(r.checkReset(name), "azurerm_postgresql_flexible_server.test"),
+			),
+		},
+	})
+}
+
+func TestAccFlexibleServerConfiguration_pgbouncerEnabled(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_postgresql_flexible_server_configuration", "test")
+	r := PostgresqlFlexibleServerConfigurationResource{}
+	name := "pgbouncer.enabled"
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.template(data, name, "true"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("name").HasValue(name),
+				check.That(data.ResourceName).Key("value").HasValue("true"),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.empty(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				data.CheckWithClientForResource(r.checkReset(name), "azurerm_postgresql_flexible_server.test"),
+			),
+		},
+	})
+}
+
+func TestAccFlexibleServerConfiguration_updateApplicationName(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_postgresql_flexible_server_configuration", "test")
+	r := PostgresqlFlexibleServerConfigurationResource{}
+	name := "application_name"
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.template(data, name, "Test APP before"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("name").HasValue(name),
+				check.That(data.ResourceName).Key("value").HasValue("Test APP before"),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.template(data, name, "Test APP after"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("name").HasValue(name),
+				check.That(data.ResourceName).Key("value").HasValue("Test APP after"),
+			),
+		},
+	})
+}
+
+func (r PostgresqlFlexibleServerConfigurationResource) checkReset(configurationName string) acceptance.ClientCheckFunc {
+	return func(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) error {
+		id, err := parse.FlexibleServerID(state.Attributes["id"])
+		if err != nil {
+			return err
+		}
+
+		resp, err := clients.Postgres.FlexibleServersConfigurationsClient.Get(ctx, id.ResourceGroup, id.Name, configurationName)
```

Comment:
Instead of using the `configurationName` to build this func, can we pull the `name` attribute from state? e.g. `state.Attributes["name"]`

Reply:
we get now configurationName from state.Attributes["name"] in tests